server/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -151,9 +167,22 @@ public void executeOrFail(DependencyCarrier dependencies,
             }
         } else {
             if (subPlan instanceof LogicalPlan logicalPlan) {
-                var planAsString = printLogicalPlan(logicalPlan, plannerContext, showCosts);
-                consumer.accept(InMemoryBatchIterator.of(new Row1(planAsString), SENTINEL), null);
-            } else if (subPlan instanceof CopyFromPlan) {
+                if (verbose) {
+                    List<RowN> rows = optimizerSteps.stream()
+                        .map(step -> {
+                            if (step.isInitial()) {
+                                return new RowN("Initial logical plan", step.planAsString());
+                            }
+                            return new RowN(step.rule().sessionSettingName(), step.planAsString());
+                        })
+                        .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
```suggestion
                        .toList();
```

Reply:
That was intentional :). I add an element to this list in the line below.